docs/Design/fpp-json-dict.md

```diff
@@ -793,40 +751,42 @@ module MyModule {
 | `frameworkVersion` | **String** representing the F´ framework version (semantic versioning) | **String** | true |
 | `projectVersion` | **String** representing the project version (semantic versioning) | **String** | true |
 | `libraryVersions` | **Array of Strings** corresponding to the version (semantic versioning) of libraries used by the F´ project | **Array of Strings** | true
-
+| `dictionarySpecVersion` | **String** representing the JSON dictionary specification version | **String** | true |
 ```json
 {
     "deploymentName": "MyDeployment",
     "frameworkVersion": "3.3.2",
     "projectVersion": "1.0.0",
-    "libraryVersions": []
+    "libraryVersions": [],
```

Comment:
As I'm wiring fpp-to-dict in to the CMake system, figuring out how to specify `libraryVersions` is giving me some troubles. I think providing an example would be helpful. Here are my questions:

1. What is the format we want to use? `libraryName=1.0.0`,  `libraryName@1.0.0` , `libraryName==1.0.0` etc... ? We probably need to add this to the spec.
2. Are we ok with only including the library name? Say the library name is very generic e.g. `arduino-i2c`, are we ok with it being impossible to figure out where the library came from unless we can look at the source project? What I'm hinting at here is we could have an optional field e.g. `remoteUrl` or something. I don't actually think it would be useful/beneficial, but I just wanted to double check with everyone.
3. How do we deal with unspecified versions? e.g. a user has created a library locally but it is not a git submodule and/or there is no version to be retrieved.

Reply:
This is a good point @thomas-bc. The format of entries in `libraryVersions` is not specified in the spec. Also, there is no check done in the fpp-to-dict tool to verify that the libraryVersions input conforms to a certain format.